src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvConfigurationsNav.tsx

```diff
@@ -0,0 +1,281 @@
+const ShimmerText = ({ width }: { width: string }) => (
+    <div className={`p-8 h-32 w-${width}`}>
+        <div className="shimmer-loading w-100 h-100" />
+    </div>
+)
+
+export const EnvConfigurationsNav = () => {
+    // HOOKS
+    const history = useHistory()
+    const { pathname } = useLocation()
+    const {
+        path,
+        params: { envId },
+    } = useRouteMatch<{ envId: string }>()
+    const { appId, resourceKind, envConfig, fetchEnvConfig, isBaseConfigProtected, environments, lastUnlockedStage } =
+        useAppConfigurationContext()
+
+    // STATES
+    const [expandedIds, setExpandedIds] = useState<Record<EnvResourceType, boolean>>()
+    const [updatedEnvConfig, setUpdatedEnvConfig] = useState<ReturnType<typeof getEnvConfiguration>>({
+        deploymentTemplate: null,
+        configMaps: [],
+        secrets: [],
+    })
+    const [jobEnvs, setJobEnvs] = useState<AppEnvironment[]>([])
+
+    // CONSTANTS
+    const isDevtronApp = resourceKind === ResourceKindType.devtronApplication
+    /** Current Environment Data. */
+    const environmentData = (isDevtronApp ? environments : jobEnvs)?.find(
+        ({ environmentId }) => environmentId === +envId,
+    ) || {
+        environmentName: 'Base Configurations',
+        environmentId: -1,
+        isProtected: isBaseConfigProtected,
+    }
+
+    const addUnnamedNavLink = (
+        _updatedEnvConfig: ReturnType<typeof getEnvConfiguration>,
+        resourceType: EnvResourceType,
+    ) =>
+        _updatedEnvConfig[resourceType === EnvResourceType.ConfigMap ? 'configMaps' : 'secrets'].push({
+            title: 'Unnamed',
+            href: getNavigationPath(
+                path,
+                appId,
+                environmentData.environmentId,
+                resourceType,
+                ResourceConfigState.Unnamed,
+            ),
+        })
+
+    useEffect(() => {
+        // Fetch the env configuration
+        fetchEnvConfig(+(envId || -1))
+
+        return () => {
+            setExpandedIds(null)
+        }
+    }, [envId])
+
+    useEffect(() => {
+        if (envConfig.config) {
+            const newEnvConfig = getEnvConfiguration(envConfig.config, path, appId, environmentData)
+
+            let resourceType: EnvResourceType
+
+            if (pathname.includes('/configmap')) {
+                resourceType = EnvResourceType.ConfigMap
+            } else if (pathname.includes('/secrets')) {
+                resourceType = EnvResourceType.Secret
+            }
+
+            if (pathname.includes('/create')) {
+                addUnnamedNavLink(newEnvConfig, resourceType)
+            }
+
+            setExpandedIds({ ...expandedIds, [resourceType]: true })
+            setUpdatedEnvConfig(newEnvConfig)
+        }
+    }, [envConfig])
+
+    useEffect(() => {
+        getJobOtherEnvironmentMin(appId)
+            .then(({ result }) => {
+                setJobEnvs(result)
+            })
+            .catch(() => {})
+    }, [])
```

Comment:
1. This call should be sent only for resource type job
2. Can we please use useAsync here instead?